syncer/syncer.go

```diff
@@ -728,11 +746,7 @@ func (s *Syncer) Connect(ctx context.Context, addr string) (*Peer, error) {
 		ConnAddr: conn.RemoteAddr().String(),
 		Inbound:  false,
 	}
-	s.wg.Add(1)
-	go func() {
-		defer s.wg.Done()
-		s.runPeer(p)
-	}()
+	go s.runPeer(p)
```

Comment:
It seems like `runPeer` also needs a threadgroup at the top.

Reply:
So initially I removed it on purpose cause I thought it's probably a bad idea to prevent `runPeer` from being executed since `Connect` always adds the peer to `s.peers`. Effectively leading to the peer to never getting removed.
But I added it and instead moved the `defer` removal in `runPeer` to the top to make sure it always runs considering that both `Connect` AND `runPeer` add the peer. So it was probably not save to do the `defer` after adding the peer in `runPeer` to begin with and might have been a potential cause for a deadlock on shutdown.